### ISSUE TITLE
New custom client, configuration problem, app keeps on stopping

### ISSUE THREAD

**Issue by deepalics0044** [e-mission/e-mission-docs#345]

**Comment by deepalics0044** [comment]
I updated all of the plugins and tested it I can't see the trips being tracked, again there is no trip started push notification.

It seems like ever since I added both  old & new client id for iOS , Android and the google.json file .I can't get through the app after registration it stops running.

For me Android 9 I can't enter into the page after user registeration. The app keeps on stopping. 

After addding the old id for iOS , the 2 step procedure is working fine for iOS.

**Comment by deepalics0044** [comment]
> And also for confirmation, which version of the data collection plugin are you on?

I used version 1.3.1

**Comment by deepalics0044** [comment]
Just for clarification

I updated all these plugins

e-mission/e-mission-transition-notify#19
e-mission/e-mission-data-collection#170
e-mission/cordova-unified-logger#29
e-mission/cordova-jwt-auth#35
e-mission/cordova-server-sync#35

by downloading them , renaming each folder name respectively for example:-

e-mission-data-collection to edu.berkeley.eecs.emission.cordova.datacollection

and adding them to plugin folder.

After adding I cordova build again.

**Comment by deepalics0044** [comment]
I made sure that google.json file stays in root directory of the project before building.

**Comment by shankari** [comment]
so what is the backtrace that you see when it "stops running".
https://opensource.guide/how-to-contribute/#communicating-effectively

> Give context. Help others get quickly up to speed. If you’re running into an error, explain what you’re trying to do and how to reproduce it. If you’re suggesting a new idea, explain why you think it’d be useful to the project (not just to you!).
>    😇 “X doesn’t happen when I do Y”
>    😢 “X is broken! Please fix it.”

**Comment by shankari** [comment]
> by downloading them , renaming each folder name respectively for example:-
> e-mission-data-collection to edu.berkeley.eecs.emission.cordova.datacollection
> and adding them to plugin folder.

I am not sure why you need to do this manually. Isn't this what `cordova prepare` does?
Also, https://github.com/e-mission/e-mission-data-collection/pull/170 is version 1.3.0

**Comment by deepalics0044** [comment]
I am at [e-mission/e-mission-data-collection#171](url)  version 1.3.1

**Comment by shankari** [comment]
Have you tried installing the beta version 
https://github.com/e-mission/e-mission-docs/issues/325#issuecomment-475348028

As I said there, it should help a lot in narrowing down the issue to platform versus your app configuration. And if it works, @ipsita0012 will also have the reassurance that you can launch on Monday even if only with the two step process.

**Comment by deepalics0044** [comment]
![Screenshot_20190322-015858](https://user-images.githubusercontent.com/40918557/54783227-d2d35e00-4c46-11e9-887e-0adcf3318ec7.jpg)

**Comment by shankari** [comment]
Hm. I thought that just sending the URL should be enough. Let me try to promote this to an alpha build.

**Comment by shankari** [comment]
ok so I tried to promote it to a beta build, which is the first version that has an open test channel, and it looks like there is a security exception on some android devices. Let me see if I can fix those and update the build...

**Comment by shankari** [comment]
@deepalics0044 @ipsita0012 any updates on this? Does the reference emTripLog work? If you want to go live on Monday, we should really focus on resolving issues, either in the platform, or in your custom build, today. So that you can get 24 hours of end to end testing before the launch

**Comment by deepalics0044** [comment]
Yeah!! The beta version works fine. The trips being tracked properly.

**Comment by shankari** [comment]
ok so then there really is a configuration issue from your side when you build logTrip, which we can walk through resolving.

@ipsita0012, you can definitely launch with at least the 2-step process, so you can get less worrried :)
I would suggest that you start end-to-end testing with the 2-step process on both android and iOS while @deepalics0044 works through fixing LogTrip so that you know that you are good 2 go on Monday.

**Comment by shankari** [comment]
have you set up the server configuration to allow login with both emTripLog and LogTrip? So that @ipsita0012 can start end to end testing with emTripLog...

**Comment by shankari** [comment]
@deepalics0044 finally, wrt to debugging LogTrip, I would start with using `cordova prepare` instead of copying the plugins manually(!!?). I am not even sure why you copied the plugins manually to be honest (and if I put this in the documentation somewhere, please let me know where so that I can fix it). `cordova prepare` does other important things to the manifest file that are key to the resulting app working properly.

**Comment by deepalics0044** [comment]
The reason behind putting the plugins manually was before I did fork the
repository (e-missin-phone)and Cordova prepare but I found out that the
plugins where not updated.

**Comment by deepalics0044** [comment]
Okay so what could be the quickest possible way to set up everything again for Logtrip?

**Comment by shankari** [comment]
@deepalics0044 @ipsita0012  first, you should set up the server to accept connections from both emTripLog and LogTrip (e.g. both `clientID`s) That way, we have at least one end-to-end solution ready to go by Monday

**Comment by deepalics0044** [comment]
I am not very sure I don't think I pull from PR. I thought after Cordova
prepare I'll have the updated plugin

**Comment by deepalics0044** [comment]
The server is accepting connection from both. I just tested.

**Comment by shankari** [comment]
> The server is accepting connection from both. I just tested.

Great! @ipsita0012, please start testing with `emTripLog` so you know that you will have a working version on Monday.

You can use http://emission.cistup.iisc.ac.in:8080/#/client_setup?new_client=cistupec&clear_usercache=true&clear_local_storage=true

On android, please test with this version of `emTripLog` instead of the one on the page:
https://play.google.com/apps/testing/edu.berkeley.eecs.embase. I will move it to production once I have the result of your testing, and the instructions on the page will be correct.

On iOS, you can use the instructions on the page unchanged.

**Comment by shankari** [comment]
> I am not very sure I don't think I pull from PR. I thought after Cordova prepare I'll have the updated 
> plugin

`cordova prepare` installs the versions that are in `config.xml` and `package.json`. The PR that updates `config.xml` and `package.json` to the latest versions has not yet been merged to master.

As I said in 
https://github.com/e-mission/e-mission-docs/issues/325#issuecomment-472662265

> @deepalics0044 in order to use dynamic permissions you can merge this PR locally
e-mission/e-mission-phone#533

you need to merge the PR to get the correct plugin versions.

Also, FYI, server seems to be down now?
http://emission.cistup.iisc.ac.in:8080/#/client_setup?new_client=cistupec&clear_usercache=true&clear_local_storage=true hangs for me...

**Comment by deepalics0044** [comment]
Instruction on the page needs to be changed ma'am. Also, we need one input
field for serial number.

**Comment by shankari** [comment]
> Instruction on the page needs to be changed ma'am. 

The instructions on the bug supercede the instructions on the page. Once the change is merged into master, the instructions on the page are correct. But you had a tight timeline, so I sent out instructions on the bug rather than waiting for the merge. If you didn't need to go live immediately, I would have made this change in a month or so instead of pushing to complete it when I was unwell.

> Also, we need one input field for serial number.

Sure, feel free to submit a PR with the new input field.

**Comment by deepalics0044** [comment]
Will changing the config.xml and package.json from e-mission-phone to that if e-mission-base will help?
I that what you mean by merge locally 

> you need to merge the PR to get the correct plugin versions.

If this is the problem I am ready to fork and change the config.xml,package.json. Build the project again. Then the only problem will be linking the survey to the app. For that I need to put a serial no. input field in the app.

**Comment by shankari** [comment]
>  I that what you mean by merge locally

No, what I mean by merge locally is:
https://stackoverflow.com/questions/48289101/merge-pull-request-from-upstream-branch-into-a-forked-repository

> Then the only problem will be linking the survey to the app. For that I need to put a serial no. input field in the app.

It is your app, and your survey. You can change anything in the cistup-ec branch, including adding a serial no. input field to the onboarding. Or you can change your survey to have a field that is automatically filled in with the user's UUID. Good luck with your changes, and remember to send me a PR so that the UI channel is also updated...

**Comment by shankari** [comment]
you can also download and apply a patch
https://stackoverflow.com/questions/28484186/apply-github-commit-pull-request-as-a-patch

**Comment by shankari** [comment]
FYI, I hadn't updated the PR on the e-mission-phone repo to 1.3.1 yet (only on the em-base repo), but I have done so now
https://github.com/e-mission/e-mission-phone/pull/533/commits/8e7eaea885ac6062a6d32bf86af9b83a52b374ea

So you should probably re-pull the PR if you have already done so.

**Comment by shankari** [comment]
> Sure, feel free to submit a PR with the new input field.

If you do go with the new input field option, I would suggest storing the resulting value in the profile, by calling `updateUser`
https://github.com/e-mission/e-mission-phone/blob/cistup-ec/www/js/services.js#L17

you can search for examples of how it is used

**Comment by deepalics0044** [comment]
Just realized the app kept on stopping after the first build because the target api version was not 26. It was higher than 26.

**Comment by deepalics0044** [comment]
Doing end to end testing for our application 'Logtrip'.

**Comment by MythriNagaraju** [comment]
this is happening with iOS too. When you made changes to update the targetSDK to level 26, did some of those changes impact the iOS app too?

**Comment by shankari** [comment]
None of the recent changes were required for iOS, and I have not published an updated version to the iOS app store since Oct.

Also the root cause here turned out to be that @deepalics0044 was copying plugins manually. I am not aware of other issues with compiling at least the android version from scratch.

**Comment by shankari** [comment]
I just tried running the iOS version of the most recent release (2.8.1). Note that this has not actually been uploaded to the iOS app store yet - the currently uploaded version is 2.7.1

And it works fine too and does not crash (see attached video). If you encounter a crash in your testing, please file a new issue with more details (version of e-mission-phone, version of iOS, physical phone/emulator....) and the stacktrace if possible and I will investigate.

**Comment by shankari** [comment]
Closing this issue to avoid confusion.

**Comment by MythriNagaraju** [comment]
Okay, I am still working on the merging 2.8.1 with my fork, but will update you on how it goes later. Thanks!
